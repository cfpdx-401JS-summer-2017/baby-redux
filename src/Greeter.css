label {
    margin-bottom: 10px;
    font-weight: bold;
}

input {
    margin-bottom: 10px;
}

.salutation {
    font-family: 'Luckiest Guy', cursive;
    font-size: 30px;
    margin-bottom: 50px;
}

.input {
    display: flex;
    flex-direction: column;
    width: 200px;
    margin: 0 auto;
}

button {
    width: 70px;
    height: 30px;
    font-size: 75%;
    background-color: black;
    color: white;
    text-transform: uppercase;
    border-radius: 5px;
    border: none;
    margin: 0 auto;
    margin-top: 20px;
}

button:focus{
    outline: none;
}