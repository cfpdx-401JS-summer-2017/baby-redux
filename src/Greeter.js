import React from 'react';
import {connect} from 'react-redux';
import {updateSalutation, updateName, reset} from './greeter.actions';

export function Greeter({salutation = '', name = '', dispatch}) {
    return (
        <div>
            <span>{salutation}, </span>
            <span>{name}</span>!
            <p>Salutation: </p>
            <input value={salutation} onChange={({target}) => dispatch(updateSalutation(target.value))}></input>
            <p>Name: </p>
            <input value={name} onChange={({target}) => dispatch(updateName(target.value))}></input>
            <br/>
            <button onClick={() => dispatch(reset())}>Reset</button>
        </div>
    )
}

export default connect(
    state => ({salutation: state.salutation, name: state.name})
)(Greeter);