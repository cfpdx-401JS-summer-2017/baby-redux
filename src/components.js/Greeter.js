import React from 'react';
import PropTypes from 'prop-types';

function Greeter() {
  Greeter.propTypes = {
    salutation: PropTypes.string,
    name: PropTypes.string
  }


  return (
    <div>
      <span>{salutation}</span>
      <span>{name}</span>!
    </div>
  );
}

export default Greeter;
