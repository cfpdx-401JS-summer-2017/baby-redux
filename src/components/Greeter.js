import React from 'react';
import { connect } from 'react-redux';
import { Editor } from './Editor';

export function Greeter({salutation, name, dispatch}) {
  console.log(salutation, name);
  return (
    <div>
      <div>
        <span>{salutation}</span>
        <span>{name}</span>!
      </div>
      <Editor salutation={salutation} name={name} dispatch={dispatch}/>
    </div>
  );
}

export default connect(
  state => ({
  salutation: state.salutation,
  name: state.name
}))(Greeter);
