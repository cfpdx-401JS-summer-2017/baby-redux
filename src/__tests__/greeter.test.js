import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Greeter } from '../Greeter';

describe('Greeter', () => {

    it('defaults to Hello Stranger', () => {
        const wrapper = shallow(<Greeter />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('Updates name', () => {
        const wrapper = shallow(<Greeter name={'Robin'}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('Updates salutation', () => {
        const wrapper = shallow(<Greeter salutation={'Hey'}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});
