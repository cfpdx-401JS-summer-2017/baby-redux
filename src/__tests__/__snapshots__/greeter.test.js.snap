// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeter Updates name 1`] = `
<div>
  <GreetingDisplay
    name="Robin"
  />
  <form>
    <label
      htmlFor="salutation"
    >
      Salutation:
    </label>
    <input
      id="salutation"
      name="salutation"
      onChange={[Function]}
    />
    <br />
    <label
      htmlFor="name"
    >
      Name:
    </label>
    <input
      id="name"
      name="name"
      onChange={[Function]}
    />
    <button
      onClick={[Function]}
      type="reset"
    >
      Reset
    </button>
  </form>
</div>
`;

exports[`Greeter Updates salutation 1`] = `
<div>
  <GreetingDisplay
    salutation="Hey"
  />
  <form>
    <label
      htmlFor="salutation"
    >
      Salutation:
    </label>
    <input
      id="salutation"
      name="salutation"
      onChange={[Function]}
    />
    <br />
    <label
      htmlFor="name"
    >
      Name:
    </label>
    <input
      id="name"
      name="name"
      onChange={[Function]}
    />
    <button
      onClick={[Function]}
      type="reset"
    >
      Reset
    </button>
  </form>
</div>
`;

exports[`Greeter defaults to Hello Stranger 1`] = `
<div>
  <GreetingDisplay />
  <form>
    <label
      htmlFor="salutation"
    >
      Salutation:
    </label>
    <input
      id="salutation"
      name="salutation"
      onChange={[Function]}
    />
    <br />
    <label
      htmlFor="name"
    >
      Name:
    </label>
    <input
      id="name"
      name="name"
      onChange={[Function]}
    />
    <button
      onClick={[Function]}
      type="reset"
    >
      Reset
    </button>
  </form>
</div>
`;
